platform/ios/app/MBXEmbeddedMapViewController.m

```diff
@@ -0,0 +1,146 @@
+#import "MBXEmbeddedMapViewController.h"
+#import <Mapbox/Mapbox.h>
+
+typedef NS_ENUM(NSInteger, MBXEmbeddedControl) {
+    MBXEmbeddedControlZoom = 0,
+    MBXEmbeddedControlScroll,
+    MBXEmbeddedControlRotation,
+    MBXEmbeddedControlPitch
+};
+
+
+@interface MBXEmbeddedMapViewController () <UIScrollViewDelegate>
+
+@property (nonatomic, strong) MGLMapView *mapView;
+
+@end
+
+@implementation MBXEmbeddedMapViewController
+
+- (void)viewDidLoad
+{
+    [super viewDidLoad];
+    
+    UIRotationGestureRecognizer *gestureRecognizer = [[UIRotationGestureRecognizer alloc] initWithTarget:self action:@selector(rotation:)];
+    
+    UIScrollView *scrollView = [[UIScrollView alloc] initWithFrame:self.view.frame];
+    scrollView.delegate = self;
+    [self.view addSubview:scrollView];
+    [scrollView addGestureRecognizer:gestureRecognizer];
+    
+    CGRect mapRect = CGRectMake(0, 0, CGRectGetWidth(scrollView.frame) * 2, CGRectGetHeight(scrollView.frame) * 2);
+    self.mapView = [[MGLMapView alloc] initWithFrame:mapRect];
```

Comment:
Since the view controller itself is already defined in the storyboard, I wonder if it would make the demo clearer if we moved some of this setup to the storyboard as well. Then most of this class’s code would relate to the switches’ behavior instead of the layout code.